components/patient-data/ui/src/main/resources/PhenoTips/UIX_Field__measurements.xml

```diff
@@ -108,51 +399,268 @@
+  &lt;div class="metric"&gt;
+    #if ($xcontext.action == 'edit')
+      #if ($handler.getComputationDependencies().size())
+        &lt;input type="hidden" name="PhenoTips.MeasurementClass_${id}_value" value="$!{val}"$disabled&gt;
+        &lt;span class="val"&gt;$!{val}&lt;/span&gt;
+      #else
+        &lt;input type="text" size="5" name="PhenoTips.MeasurementClass_${id}_value" value="$!{val}"$disabled&gt;
       #end
-    #end## foreach object
-    ##
-  #end## if dataclass
+      &lt;input type="hidden" name="PhenoTips.MeasurementClass_${id}_age" value=""$disabled&gt;
+      &lt;input type="hidden" name="PhenoTips.MeasurementClass_${id}_date" value=""$disabled&gt;
+      &lt;input type="hidden" name="PhenoTips.MeasurementClass_${id}_side" value="$!{side}"$disabled&gt;
+    #else
+      &lt;span class="val"&gt;$!{val}&lt;/span&gt;
+    #end
+    &lt;span class="unit"&gt;#if ("$!unit" != "")$services.localization.render("phenotips.UIXField.measurements.units.${unit}")#end&lt;/span&gt;
+  &lt;/div&gt;
+  &lt;span class="pctl#if ("$!val" != "") $services.measurements.getFuzzyValue($sd)#end"&gt;#if ("$!val" != "")$eval#end&lt;/span&gt;
+  {{/html}}
+)))
+#end##
+##
+## Source: http://extensions.xwiki.org/xwiki/bin/view/Extension/Sort+a+list+of+XWiki+objects
+#macro (sortXWikiObjectsList $inputObjects $propertyName $outputObjects)
+  #set ($objectsMap = {})
+  #foreach ($object in $inputObjects)
+    #set ($key = $object.getProperty($propertyName).value)
+    #set ($sameKeyObjects = $objectsMap.get($key))
+    #if (!$sameKeyObjects)
+      #set ($sameKeyObjects = [])
+      #set ($discard = $objectsMap.put($key, $sameKeyObjects))
+    #end
+    #set ($discard = $sameKeyObjects.add($object))
+  #end
+  #foreach ($key in $sorttool.sort($objectsMap.keySet()))
+    #set ($sameKeyObjects = $objectsMap.get($key))
+    #foreach ($sameKeyObject in $sameKeyObjects)
+      #set ($discard = $outputObjects.add($sameKeyObject))
+    #end
+  #end
+#end##
+##
+#macro (measurements $showDefault $showEnabled $measurementSet $proto)
+  #set ($orderedMeasurements = $configObj.getProperty('ordering').value)
+  #foreach ($measurement in $orderedMeasurements)
+    #set ($handler = $services.measurements.get($measurement))
+    #if (($showEnabled == $configObj.getProperty('enabled').value.contains($handler.getName()))
+         &amp;&amp; ($showDefault == $configObj.getProperty('default').value.contains($handler.getName())))
+      #set($name = $handler.getName())
+      #if ($handler.isDoubleSided())
+        #set($vals = {})
+        #set($discard = $vals.put("r", $measurementSet.get("${name}-r")))
+        #set($discard = $vals.put("l", $measurementSet.get("${name}-l")))
+      #elseif ($measurementSet.get($name))
+        #set($vals = $measurementSet.get($name))
+      #else
+        #set($vals = {})
+      #end
+      #measurement($handler $vals $proto)
+    #end
+  #end
+#end##
+##
+#macro (measurementSet $set)
+#if ("$!set" == "")
```

Comment:
Sometimes there is a indentation right after the ```macro``` declaration (like above) and sometimes not